### ISSUE TITLE
Bullet proof wrings

### ISSUE THREAD

**Issue by yossigil** [SpartanRefactoring/Spartanizer#196]
Even if a wring throws an exception, you continue.

**Comment by yossigil** [comment]
@M4tteo : this is what we need for the experiments.

**Comment by yossigil** [comment]
See also #197 
Whenever one throws an exception record it using java.util.logging. Go over the entire code base.

**Comment by yossigil** [comment]
Provide a mechanism to record in a separate file cases where tippers throw an exception

**Comment by OriRoth** [comment]
Done (?). I have not committed example logs because they are too big.
*Instructions*:
* If you want to log the tests:
  * Create a directory called `logs` in `Spartanize` (main directory)
  * Run tests
  * It is not recommended to commit those files, beware!
* If you want to log in debug mode / release (?):
  * Create a directory called `logs` in eclipse's installation directory (i.e. where `eclipse.exe` is)
  * Run eclipse as usual

@M4tt30rru you may consider adding this mechanism to you cmd tool (?) see `Trimmer` for usage example.

**Comment by OriRoth** [comment]
Logging format:
<exception's stack trace>
<file name>
<whole code>
<code currently inspected>

Notice "file name" is *not* available for tests, so we get a "???" instead.
@yossigil anything else? something to change? is it working on your machine?

**Comment by yossigil** [comment]
Add these files to `.gitignore`

**Comment by yossigil** [comment]
Of course, it would be an amazing project to convert these XML files to unit tets, or better yet, use advanced testing / junit that would run everything thus collected.

**Comment by OriRoth** [comment]
OK I made something nice...

See `Autamatic_2016_12_14_01_52_17` test class in `il.org.spartan.automatic` package under tests. This test class was automatically generated (I did not wrote a thing!) by a tool I built (`LogToTest`).

*What I did*:
* Ran `il.org.spartan.spartanizer.research` test package as a JUnit test
  * File `log_spartan_2016-12-14-01-52-05.txt` appeared in `logs` (not committed)
* Ran `LogToTest` as a Java application (without any command line parameters)
  * This tool enters the `logs` directory, collect logs and traverse error messages
  * Creates tests only for new problems
  * New test file named `Autamatic_2016_12_14_01_52_17` appeared in `automatic` tests package
    * I only formatted the code

*What I got*:
* From a 2599 lines of log file I created a laconic test file containing only 5 tests
* The tests are in the form of `trimmingOf(...).doesNotCrash();`. `doesNotCrash` is a new addition I wrote for `Operand` that simply checks spartanization of given code does not crash (works... for most cases, see #950)

**Comment by yossigil** [comment]
Amazing job. When would you take a break?
